src/main/java/de/tum/in/www1/artemis/service/NotificationSettingsService.java

```diff
@@ -0,0 +1,84 @@
+package de.tum.in.www1.artemis.service;
+
+import static de.tum.in.www1.artemis.domain.enumeration.NotificationType.*;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.NotificationOption;
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.domain.enumeration.NotificationType;
+import de.tum.in.www1.artemis.domain.notification.NotificationTitleTypeConstants;
+
+@Service
+public class NotificationSettingsService {
+
+    /**
+     * This is the place where the mapping between OptionSpecifiers and NotificationTypes happens on the server side
+     * Each OptionSpecifiers can be based on multiple different NotificationTypes
+     */
+    private final static Map<String, NotificationType[]> NOTIFICATION_OPTION_SPECIFIER_TO_NOTIFICATION_TYPES_MAP = Map.of(
+            "notification.exercise-notification.exercise-created-or-started", new NotificationType[] { EXERCISE_CREATED },
+            "notification.exercise-notification.exercise-open-for-practice", new NotificationType[] { EXERCISE_PRACTICE }, "notification.exercise-notification.new-post-exercises",
+            new NotificationType[] { NEW_POST_FOR_EXERCISE }, "notification.exercise-notification.new-answer-post-exercises",
+            new NotificationType[] { NEW_ANSWER_POST_FOR_EXERCISE }, "notification.lecture-notification.attachment-changes", new NotificationType[] { ATTACHMENT_CHANGE },
```

Comment:
[nit] this doesn't look very prettified 😄 . or does the commit hook automatically keep the line long like this. maybe keep them one entry per line, looks more readable.

Reply:
I see, smth happened during commiting/pushing ...

Reply:
I edited it in gitHub directly so now it should be fine^^

Reply:
This lead to codacy issues so I looked at the code locally again, made the style as it should (each line one entry) and even executen prettier on it but it again changed it during the commiting/pushing to this ... strange

Reply:
I have now changed the initialization of the map. This should do the trick